### DISCUSSION TITLE
How to compare AFL++ 2.65c(AFL++ in invscov) and 4.00c(AFL++) for exiv2 Testing: Analyzing Differences in edges_found and bitmap_cvg

### DISCUSSION THREAD

**Discussion by JohenanLi** [AFLplusplus/AFLplusplus#1745] [Q&A]
In version 2.65c, there are significantly more **edges_found** and the **bitmap_cvg** is also larger. However, after examining some of the source code, I found that the calculation of bitmap_cvg is different. I would like to convert them, but I'm unsure how to proceed. Below are the fuzzer_stats data for both tests.
If anyone knows how to do it, please let me know. Thank you.
2.65c:
fuzzer_pid        : 2230351
cycles_done       : 16
cycles_wo_finds   : 0
execs_done        : 3209805
execs_per_sec     : 37.15
paths_total       : 5546
paths_favored     : 612
paths_found       : 356
paths_imported    : 0
max_depth         : 10
cur_path          : 621
pending_favs      : 0
pending_total     : 3574
variable_paths    : 0
stability         : 100.00%
bitmap_cvg        : 45.73%
unique_crashes    : 65
unique_hangs      : 11
last_path         : 1683688745
last_crash        : 1683688694
last_hang         : 1683688177
execs_since_crash : 11409
exec_timeout      : 420
slowest_exec_ms   : 0
peak_rss_mb       : 0
edges_found       : 29969
var_byte_count    : 0
afl_banner        : exiv2
afl_version       : ++2.65c
target_mode       : default

4.00c:
start_time        : 1682585639
last_update       : 1682672039
run_time          : 86400
fuzzer_pid        : 296262
cycles_done       : 1
cycles_wo_finds   : 0
execs_done        : 17314015
execs_per_sec     : 200.39
execs_ps_last_min : 211.47
corpus_count      : 5713
corpus_favored    : 496
corpus_found      : 523
corpus_imported   : 0
corpus_variable   : 0
max_depth         : 10
cur_item          : 823
pending_favs      : 0
pending_total     : 4285
stability         : 100.00%
bitmap_cvg        : 9.75%
saved_crashes     : 118
saved_hangs       : 17
last_find         : 1682671962
last_crash        : 1682669550
last_hang         : 1682647127
execs_since_crash : 490668
exec_timeout      : 40
slowest_exec_ms   : 0
peak_rss_mb       : 0
cpu_affinity      : 0
edges_found       : 7049
total_edges       : 72276
var_byte_count    : 0
havoc_expansion   : 0
auto_dict_entries : 0
testcache_size    : 49842765
testcache_count   : 3362
testcache_evict   : 2776448
afl_banner        : .../build_instrument/bin/exiv2
afl_version       : ++4.00c
target_mode       : shmem_testcase default

**Comment by vanhauser-thc** [comment]
you cannot directly compare them. different llvm version plus different afl++ instrument differently and hence edges are different. newer versions should have more edges

also afl++ 2.65c does not have colliding free coverage so there is no correlation what so ever unless you use the outdated CLASSIC mode in current afl++.

It is as if you would want to compare Word 6.0 vs Office 2013 ...
